src/Nethermind/Nethermind.Serialization.Rlp/Rlp.cs

```diff
@@ -407,60 +407,20 @@ public static Rlp Encode(ulong value)
 
         public static Rlp Encode(long value)
         {
-            if (value == 0L)
-            {
-                return OfEmptyByteArray;
-            }
-
-            if (value > 0)
-            {
-                byte byte6 = (byte) (value >> 8);
-                byte byte5 = (byte) (value >> 16);
-                byte byte4 = (byte) (value >> 24);
-                byte byte3 = (byte) (value >> 32);
-                byte byte2 = (byte) (value >> 40);
-                byte byte1 = (byte) (value >> 48);
-                byte byte0 = (byte) (value >> 56);
-
-                if (value < 256L * 256L * 256L * 256L * 256L * 256L * 256L)
-                {
-                    if (value < 256L * 256L * 256L * 256L * 256L * 256L)
-                    {
-                        if (value < 256L * 256L * 256L * 256L * 256L)
-                        {
-                            if (value < 256L * 256L * 256L * 256L)
-                            {
-                                if (value < 256 * 256 * 256)
-                                {
-                                    if (value < 256 * 256)
-                                    {
-                                        if (value < 128)
-                                        {
-                                            return new Rlp((byte) value);
-                                        }
-
-                                        return value < 256 ? new Rlp(new byte[] {129, (byte) value}) : new Rlp(new byte[] {130, byte6, (byte) value});
-                                    }
-
-                                    return new Rlp(new byte[] {131, byte5, byte6, (byte) value});
-                                }
-
-                                return new Rlp(new byte[] {132, byte4, byte5, byte6, (byte) value});
-                            }
-
-                            return new Rlp(new byte[] {133, byte3, byte4, byte5, byte6, (byte) value});
-                        }
-
-                        return new Rlp(new byte[] {134, byte2, byte3, byte4, byte5, byte6, (byte) value});
-                    }
-
-                    return new Rlp(new byte[] {135, byte1, byte2, byte3, byte4, byte5, byte6, (byte) value});
-                }
-
-                return new Rlp(new byte[] {136, byte0, byte1, byte2, byte3, byte4, byte5, byte6, (byte) value});
-            }
-
-            return Encode(new BigInteger(value), 8);
+            return value switch
+            {
+                < 0 => Encode(new BigInteger(value), 8),
```

Comment:
is the order having an impact on performance? Before this was last.

Reply:
Yes the order does affect performance it seams. Before adding the first two ones we had a small Perf improvement in some scenarios

Reply:
Can we move it to the end?

Reply:
Updated with the benchmarks. I had run the benchmarks incorrectly before. now they reflect a slight improvement. But with your suggestion to have <0 at the bottom, it had worse performance. you can see that in improvedNew. 
In conclusion the current implementation is the best one.